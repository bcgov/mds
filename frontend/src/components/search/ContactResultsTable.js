import React from "react";
import { Table, Row, Col, Divider } from "antd";
import PropTypes from "prop-types";
import Highlight from "react-highlighter";

import { Link } from "react-router-dom";
import * as router from "@/constants/routes";

/**
 * @class  ContactResultsTable - displays a table of mine search results
 */

const propTypes = {
  header: PropTypes.string.isRequired,
  highlightRegex: PropTypes.string.isRequired,
  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
  query: PropTypes.string.isRequired,
  partyRelationshipTypeHash: PropTypes.objectOf(PropTypes.strings).isRequired,
};

const defaultProps = {};

const parseQuery = (query) => {
  const exp = RegExp("@");
  if (exp.test(query)) {
    return { email: query };
  }
  const terms = query.split(" ");
  if (terms.length === 1) {
    return { last_name: terms[0] };
  }
  return { first_name: terms[0], last_name: terms[1] };
};

export const ContactResultsTable = (props) => {
  const columns = [
    {
      title: "Party Guid",
      dataIndex: "party_guid",
      key: "party_guid",
      render: (text, record) => [
        <Row>
          <Col span={24}>
            <Link
              to={router.PARTY_PROFILE.dynamicRoute(record.party_guid)}
              style={{ fontSize: "1.5rem", fontWeight: "bold" }}
            >
              <Highlight search={props.highlightRegex}>{record.name}</Highlight>
            </Link>
          </Col>
        </Row>,
        <Row style={{ paddingTop: "5px" }}>
          <Col xs={24} md={4}>
            <p>Roles</p>
          </Col>
          <Col xs={24} md={8}>
            <p>
              {props.partyRelationshipTypeHash.PMT &&
                record.mine_party_appt.map((pr) => [
                  <span>{props.partyRelationshipTypeHash[pr.mine_party_appt_type_code]}</span>,
                  <br />,
                ])}
            </p>
          </Col>
          <Col xs={24} md={4}>
            <p>Email</p>
          </Col>
          <Col xs={24} md={8}>
            <p>
              <Highlight search={props.highlightRegex}>{record.email}</Highlight>
            </p>
          </Col>
          <Col xs={24} md={4}>
            <p>Mine</p>
          </Col>
          <Col xs={24} md={8}>
            <p>
              <Highlight search={props.highlightRegex}>
                {record.mine_party_appt[0].mine.mine_name}
              </Highlight>
            </p>
          </Col>
          <Col xs={24} md={4}>
            <p>Phone</p>
          </Col>
          <Col xs={24} md={8}>
            <p>
              <Highlight search={props.highlightRegex}>{record.phone_no}</Highlight>
              {record.phone_ext && record.phone_ext}
            </p>
          </Col>
        </Row>,
      ],
    },
  ];
  return (
    <div>
      <h2>{props.header}</h2>
      <Divider />
      <Table
        className="nested-table"
        align="left"
        showHeader={false}
        pagination={false}
        columns={columns}
        dataSource={props.searchResults}
      />
      <Link to={router.CONTACT_HOME_PAGE.dynamicRoute(parseQuery(props.query))}>
        <p style={{ fontSize: "22px", color: "inherit", float: "right" }}>
          See all results in Contacts ({props.searchResults.length})
        </p>
      </Link>
    </div>
  );
};

ContactResultsTable.propTypes = propTypes;
ContactResultsTable.defaultProps = defaultProps;

export default ContactResultsTable;
