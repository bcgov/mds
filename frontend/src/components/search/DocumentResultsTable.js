import React from "react";
import { Table, Row, Col, Divider } from "antd";
import PropTypes from "prop-types";
import Highlight from "react-highlighter";
import downloadFileFromDocumentManager from "@/utils/actionlessNetworkCalls";
import LinkButton from "@/components/common/LinkButton";

/**
 * @class  DocumentResultsTable - displays a table of mine search results
 */

const propTypes = {
  header: PropTypes.string.isRequired,
  highlightRegex: PropTypes.objectOf(PropTypes.regexp).isRequired,
  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const defaultProps = {};

export const DocumentResultsTable = (props) => {
  const columns = [
    {
      title: "Document Guid",
      dataIndex: "document_guid",
      key: "document_guid",
      render: (text, record) => [
        <Row>
          <Col span={24}>
            <LinkButton
              key={record.mine_document_guid || record.permit_amendment_document_guid}
              onClick={() => downloadFileFromDocumentManager(record)}
            >
              <Highlight search={props.highlightRegex}>{record.document_name}</Highlight>
            </LinkButton>
          </Col>
        </Row>,
        <Row className="padding-small--top">
          <Col xs={24} md={6}>
            <p>Mine</p>
          </Col>
          <Col xs={24} md={18}>
            <p>{record.mine_name}</p>
          </Col>
        </Row>,
      ],
    },
  ];
  return (
    <div>
      <h2>{props.header}</h2>
      <Divider />
      <Table
        className="nested-table padding-large--bottom"
        align="left"
        showHeader={false}
        pagination={false}
        columns={columns}
        dataSource={props.searchResults}
      />
    </div>
  );
};

DocumentResultsTable.propTypes = propTypes;
DocumentResultsTable.defaultProps = defaultProps;

export default DocumentResultsTable;
