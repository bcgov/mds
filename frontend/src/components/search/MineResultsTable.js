import React from "react";
import { Table, Row, Col, Divider } from "antd";
import PropTypes from "prop-types";
import Highlight from "react-highlighter";
import { Link } from "react-router-dom";
import * as router from "@/constants/routes";

/**
 * @class  MineResultsTable - displays a table of mine search results
 */

const propTypes = {
  header: PropTypes.string.isRequired,
  highlightRegex: PropTypes.objectOf(PropTypes.Any).isRequired,
  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const defaultProps = {};

export const MineResultsTable = (props) => {
  const columns = [
    {
      title: "Mine Guid",
      dataIndex: "mine_guid",
      key: "mine_guid",
      render: (text, record) => [
        <Row>
          <Col span={24}>
            <Link to={router.MINE_SUMMARY.dynamicRoute(record.mine_guid)}>
              <p style={{ fontSize: "22px", color: "inherit" }}>
                <Highlight search={props.highlightRegex}>{record.mine_name}</Highlight>
              </p>
            </Link>
          </Col>
        </Row>,
        <Row style={{ paddingTop: "5px" }}>
          <Col xs={24} md={4}>
            <p>Mine No.:</p>
          </Col>
          <Col xs={24} md={4}>
            <p>
              <Highlight search={props.highlightRegex}>{record.mine_no}</Highlight>
            </p>
          </Col>
          <Col xs={24} md={4}>
            <p>Permit No.:</p>
          </Col>
          <Col xs={24} md={12}>
            <p>{record.mine_permit.map((permit) => [<span>{permit.permit_no}</span>, <br />])}</p>
          </Col>
        </Row>,
        <Row style={{ paddingTop: "5px", paddingBottom: "15px" }}>
          <Col xs={24} md={4}>
            <p>Region:</p>
          </Col>
          <Col xs={24} md={4}>
            <p>{record.mine_region}</p>
          </Col>
          <Col xs={24} md={4}>
            <p>Status:</p>
          </Col>
          <Col xs={24} md={12}>
            <p>{record.mine_status[0] && record.mine_status[0].status_labels.join(", ")}</p>
          </Col>
        </Row>,
      ],
    },
  ];
  return (
    <div>
      <h2>{props.header}</h2>
      <Divider />
      <Table
        className="nested-table"
        align="left"
        showHeader={false}
        pagination={false}
        columns={columns}
        dataSource={props.searchResults}
      />
    </div>
  );
};

MineResultsTable.propTypes = propTypes;
MineResultsTable.defaultProps = defaultProps;

export default MineResultsTable;
