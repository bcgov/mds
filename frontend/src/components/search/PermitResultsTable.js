import React from "react";
import { Table, Row, Col, Divider } from "antd";
import PropTypes from "prop-types";
import Highlight from "react-highlighter";
import { Link } from "react-router-dom";
import * as router from "@/constants/routes";

/**
 * @class  PermitResultsTable - displays a table of mine search results
 */

const propTypes = {
  header: PropTypes.string.isRequired,
  highlightRegex: PropTypes.string.isRequired,
  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const defaultProps = {};

export const PermitResultsTable = (props) => {
  const columns = [
    {
      title: "Permit Guid",
      dataIndex: "permit_guid",
      key: "permit_guid",
      render: (text, record) => [
        <Row style={{ paddingBottom: "5px" }}>
          <Col span={12}>
            <Link to={router.MINE_SUMMARY.dynamicRoute(record.mine_guid, "permit")}>
              <p style={{ fontSize: "22px", color: "inherit" }}>
                <Highlight search={props.highlightRegex}>{record.permit_no}</Highlight>
              </p>
            </Link>
          </Col>
        </Row>,
        <Row style={{ paddingTop: "5px" }}>
          <Col span={4}>Permitee</Col>
          <Col span={6}>
            <Highlight search={props.highlightRegex}>{record.permitee}</Highlight>
          </Col>
        </Row>,
        <Row style={{ paddingTop: "5px" }}>
          <Col span={4}>Mine</Col>
          <Col span={10}>{record.mine_name}</Col>
        </Row>,
      ],
    },
  ];

  return (
    <Col md={12} sm={24} style={{ padding: "30px", paddingBottom: "60px" }}>
      <h2>{props.header}</h2>
      <Divider />
      <Table
        className="nested-table"
        align="left"
        showHeader={false}
        pagination={false}
        columns={columns}
        dataSource={props.searchResults}
      />
    </Col>
  );
};

PermitResultsTable.propTypes = propTypes;
PermitResultsTable.defaultProps = defaultProps;

export default PermitResultsTable;
