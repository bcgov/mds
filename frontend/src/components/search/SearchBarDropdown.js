import React from "react";
import PropTypes from "prop-types";
import { Menu, Icon } from "antd";
import * as route from "@/constants/routes";

import { MINE, TEAM } from "@/constants/assets";

const propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  searchTerm: PropTypes.string.isRequired,
  searchTermHistory: PropTypes.arrayOf(PropTypes.string).isRequired,
  searchBarResults: PropTypes.arrayOf(PropTypes.any).isRequired,
};

const staticMenuItems = [
  <Menu.ItemGroup title="Lookup">
    <Menu.Item key="/dashboard/mines">
      <p>
        <img
          alt="Mine"
          className="padding-small--right vertical-align-sm icon-svg-filter"
          height="25px"
          src={MINE}
        />{" "}
        Mines
      </p>
    </Menu.Item>
    ,
    <Menu.Item key="/dashboard/contacts">
      <p>
        <img
          alt="team"
          className="padding-small--right vertical-align-sm icon-svg-filter"
          height="25px"
          src={TEAM}
        />{" "}
        Contacts
      </p>
    </Menu.Item>
  </Menu.ItemGroup>,
];

const URLFor = (item) =>
  ({
    mine: route.MINE_GENERAL.dynamicRoute(item.result.id),
    party: route.PARTY_PROFILE.dynamicRoute(item.result.id),
    permit: route.MINE_PERMITS.dynamicRoute(item.result.id),
  }[item.type]);

export const SearchBarDropdown = (props) => (
  <Menu
    onMouseDown={(e) => {
      e.target.click(e);
    }}
    onClick={({ key }) => props.history.push(key)}
    selectable={false}
  >
    {props.searchTerm.length && props.searchBarResults.length
      ? [
          props.searchBarResults.map((item) => (
            <Menu.Item key={URLFor(item)}>
              <p>{`${item.result.value || ""}`}</p>
            </Menu.Item>
          )),
          <Menu.Divider />,
          <Menu.Item key={`/search?q=${props.searchTerm}`}>
            <h6>{`See all results for "${props.searchTerm}"`}</h6>
          </Menu.Item>,
        ]
      : [
          staticMenuItems,
          props.searchTermHistory.length && [
            <Menu.Divider />,
            <Menu.ItemGroup title="Recent searches">
              {props.searchTermHistory.map((pastSearchTerm) => (
                <Menu.Item key={`/search?q=${pastSearchTerm}`}>
                  <p style={{ fontStyle: "italic" }}>
                    <Icon type="search" /> {pastSearchTerm}
                  </p>
                </Menu.Item>
              ))}
            </Menu.ItemGroup>,
          ],
        ]}
  </Menu>
);

SearchBarDropdown.propTypes = propTypes;

export default SearchBarDropdown;
