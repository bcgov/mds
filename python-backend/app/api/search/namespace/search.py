from flask_restplus import Namespace

from ..search.resources.search import SearchResource, SearchOptionsResource
from app.api.search.search.resources.simple_search import SimpleSearchResource

api = Namespace('search', description='Search related operations')

api.add_resource(SearchResource, '')
api.add_resource(SearchOptionsResource, '/options')
api.add_resource(SimpleSearchResource, '/simple')