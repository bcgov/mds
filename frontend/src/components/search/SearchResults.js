import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import queryString from "query-string";
import { Row, Col } from "antd";
import { getSearchResults, getSearchTerms } from "@/selectors/searchSelectors";
import { MineResultsTable } from "@/components/search/MineResultsTable";
import { PermitResultsTable } from "@/components/search/PermitResultsTable";
import { ContactResultsTable } from "@/components/search/ContactResultsTable";
import { DocumentResultsTable } from "@/components/search/DocumentResultsTable";
import { getPartyRelationshipTypeHash } from "@/selectors/partiesSelectors";
import { fetchPartyRelationshipTypes } from "@/actionCreators/partiesActionCreator";
import { fetchSearchOptions, fetchSearchResults } from "@/actionCreators/searchActionCreator";
import Loading from "@/components/common/Loading";
import LinkButton from "@/components/common/LinkButton";
import _ from "lodash";
import { getSearchOptions } from "../../reducers/searchReducer";

const propTypes = {
  location: PropTypes.shape({ search: PropTypes.string }).isRequired,
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  fetchSearchOptions: PropTypes.func.isRequired,
  fetchSearchResults: PropTypes.func.isRequired,
  searchOptions: PropTypes.arrayOf(PropTypes.any).isRequired,
  searchResults: PropTypes.objectOf(PropTypes.any),
  searchTerms: PropTypes.arrayOf(PropTypes.string),
  partyRelationshipTypeHash: PropTypes.objectOf(PropTypes.strings).isRequired,
  fetchPartyRelationshipTypes: PropTypes.func.isRequired,
};

const defaultProps = {
  searchResults: [],
  searchTerms: [],
};

const renderSearchResultGroup = (group, searchTerms, partyRelationshipTypeHash, queryParams) => {
  const highlightRegex = RegExp(`${searchTerms.join("|")}`, "i");
  const query = queryString.parse(queryParams);
  if (group.type === "mine") {
    return (
      <Col sm={24} lg={12} style={{ padding: "30px", paddingBottom: "60px" }}>
        <MineResultsTable
          header="Mines"
          highlightRegex={highlightRegex}
          searchResults={group.results}
          query={query.q}
        />
      </Col>
    );
  }
  if (group.type === "permit") {
    return (
      <Col sm={24} lg={12} style={{ padding: "30px", paddingBottom: "60px" }}>
        <PermitResultsTable
          header="Permits"
          highlightRegex={highlightRegex}
          searchResults={group.results}
        />
      </Col>
    );
  }
  if (group.type === "party") {
    return (
      <Col sm={24} lg={12} style={{ padding: "30px", paddingBottom: "60px" }}>
        <ContactResultsTable
          header="Contacts"
          highlightRegex={highlightRegex}
          searchResults={group.results}
          partyRelationshipTypeHash={partyRelationshipTypeHash}
          query={query.q}
        />
      </Col>
    );
  }
  if (group.type === "permit_documents" || group.type === "mine_documents") {
    return (
      <Col sm={24} lg={12} style={{ padding: "30px", paddingBottom: "60px" }}>
        <DocumentResultsTable
          header={group.type === "permit_documents" ? "Permit Documents" : "Mine Documents"}
          highlightRegex={highlightRegex}
          searchResults={group.results}
        />
      </Col>
    );
  }
  return <div />;
};

export class SearchResults extends Component {
  state = {
    isSearching: false,
    hasSearchTerm: false,
    params: {},
  };

  componentDidMount = () => {
    if (!this.props.partyRelationshipTypeHash.PMT) {
      this.props.fetchPartyRelationshipTypes();
    }
    if (!this.props.searchOptions.length) {
      this.props.fetchSearchOptions();
    }
    this.handleSearch(this.props.location);
  };

  componentWillReceiveProps = (nextProps) => {
    const locationChanged = nextProps.location !== this.props.location;
    if (locationChanged) {
      this.handleSearch(nextProps.location);
    }

    // The intention below is to trigger every time the object is updated, even if the
    // array contents are the same, therefore compare with != vs !==
    // eslint-disable-next-line eqeqeq
    if (nextProps.searchResults != this.props.searchResults) {
      this.setState({ isSearching: false });
    }
  };

  handleSearch = (location) => {
    const parsedParams = queryString.parse(location.search);
    const { q, t } = parsedParams;

    if (q) {
      this.props.fetchSearchResults(q, t);
      this.setState({ params: parsedParams, isSearching: true, hasSearchTerm: true });
    }
  };

  render = () => {
    const groupedSearchResults = [];
    Object.entries(this.props.searchResults).forEach((entry) => {
      const key = entry[0];
      const value = entry[1];
      groupedSearchResults.push({
        type: key,
        score: _.sumBy(value, "score"),
        results: _.map(value, "result"),
      });
    });

    groupedSearchResults.sort((a, b) => a.score - b.score);
    groupedSearchResults.reverse();

    return (
      this.state.hasSearchTerm && (
        <div className="landing-page">
          {this.state.isSearching ? (
            <Loading />
          ) : (
            <div>
              <div className="landing-page__header">
                <h1 style={{ paddingBottom: "5px" }}>
                  Search results for {this.props.searchTerms.map((t) => `"${t}"`).join(", ")}
                </h1>
                <div>
                  Just show me:
                  {this.props.searchOptions.map((o) => (
                    <span style={{ padding: "20px" }}>
                      <LinkButton
                        key={o.model_id}
                        onClick={() => {
                          this.props.history.push(
                            `/search?q=${this.state.params.q}&t=${o.model_id}`
                          );
                        }}
                      >
                        {o.description}
                      </LinkButton>
                    </span>
                  ))}
                </div>
              </div>
              <div className="landing-page__content">
                <div className="tab__content">
                  {groupedSearchResults.length === 0 && [
                    <br />,
                    <h2>No Results Found</h2>,
                    <p>Please try another search.</p>,
                  ]}
                  <Row gutter={10}>
                    {groupedSearchResults.map((group) =>
                      renderSearchResultGroup(
                        group,
                        this.props.searchTerms,
                        this.props.partyRelationshipTypeHash,
                        this.props.location.search
                      )
                    )}
                  </Row>
                </div>
              </div>
            </div>
          )}
        </div>
      )
    );
  };
}

const mapStateToProps = (state) => ({
  searchOptions: getSearchOptions(state),
  searchResults: getSearchResults(state),
  searchTerms: getSearchTerms(state),
  partyRelationshipTypeHash: getPartyRelationshipTypeHash(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchPartyRelationshipTypes,
      fetchSearchOptions,
      fetchSearchResults,
    },
    dispatch
  );

SearchResults.propTypes = propTypes;
SearchResults.defaultProps = defaultProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchResults);
