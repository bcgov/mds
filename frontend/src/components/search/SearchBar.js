import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Input, Dropdown, Card, Col } from "antd";
import { getSearchBarResults } from "@/selectors/searchSelectors";
import { fetchSearchBarResults, clearSearchBarResults } from "@/actionCreators/searchActionCreator";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import * as router from "@/constants/routes";
import { SearchBarDropdown } from "@/components/search/SearchBarDropdown";
import { throttle } from "lodash";

const { Search } = Input;

const propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  fetchSearchBarResults: PropTypes.func.isRequired,
  clearSearchBarResults: PropTypes.func.isRequired,
  searchBarResults: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const defaultPlaceholderText = "Search";
const selectedPlaceholderText = "Type to search for mines, contacts, or more";

export class SearchBar extends Component {
  state = {
    isSelected: false,
    searchTerm: "",
    searchTermHistory: [],
  };

  constructor(props) {
    super(props);
    this.fetchSearchResultsThrottled = throttle(this.props.fetchSearchBarResults, 2000, {
      leading: true,
      trailing: true,
    });
  }

  changeSearchTerm = (e) => {
    const newSearchTerm = e.target.value;
    this.setState({ searchTerm: newSearchTerm });

    if (newSearchTerm.length > 2) {
      this.fetchSearchResultsThrottled(newSearchTerm);
    }
  };

  search = (searchTerm) => {
    if (searchTerm) {
      this.setState((prevState) => {
        const newSearchTermHistory = prevState.searchTermHistory.slice(
          Math.max(prevState.searchTermHistory.length - 2, 0)
        );
        newSearchTermHistory.push(searchTerm);
        return {
          isSelected: false,
          searchTermHistory: newSearchTermHistory,
        };
      });
      this.clearSearchBar(false);
      this.props.history.push(router.SEARCH_RESULTS.dynamicRoute({ q: searchTerm }));
    }
  };

  clearSearchBar = (isSelected = true) => {
    this.props.clearSearchBarResults();
    this.setState({
      isSelected,
      searchTerm: "",
    });
  };

  render = () => (
    <Col
      id="searchBox"
      md={4}
      lg={6}
      xl={8}
      style={{
        marginLeft: "auto",
        marginTop: "auto",
        marginBottom: "auto",
        paddingRight: "20px",
      }}
    >
      <Dropdown
        overlay={
          <Card>
            <SearchBarDropdown
              history={this.props.history}
              searchTerm={this.state.searchTerm}
              searchTermHistory={this.state.searchTermHistory}
              searchBarResults={this.props.searchBarResults}
            />
          </Card>
        }
        getPopupContainer={() => document.getElementById("searchBox")}
        trigger={[""]}
        visible={this.state.isSelected}
      >
        <Search
          size="large"
          value={this.state.searchTerm}
          placeholder={this.state.isSelected ? selectedPlaceholderText : defaultPlaceholderText}
          onSearch={(searchTerm) => this.search(searchTerm)}
          onChange={this.changeSearchTerm}
          onClick={this.clearSearchBar}
          onFocus={this.clearSearchBar}
          onBlur={() =>
            this.setState({
              isSelected: false,
            })
          }
        />
      </Dropdown>
    </Col>
  );
}

const mapStateToProps = (state) => ({
  searchBarResults: getSearchBarResults(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchSearchBarResults,
      clearSearchBarResults,
    },
    dispatch
  );

SearchBar.propTypes = propTypes;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(SearchBar));
