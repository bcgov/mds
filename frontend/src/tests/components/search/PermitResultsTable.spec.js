import React from "react";
import { shallow } from "enzyme";
import { PermitResultsTable } from "@/components/search/PermitResultsTable";
import * as MOCK from "@/tests/mocks/dataMocks";

const reducerProps = {};

const setupReducerProps = () => {
  reducerProps.header = MOCK.SEARCH_OPTIONS[2].description;
  reducerProps.highlightRegex = RegExp(".*");
  reducerProps.searchResults = MOCK.PERMIT_SEARCH_RESULTS;
};

beforeEach(() => {
  setupReducerProps();
});

describe("PermitResultsTable", () => {
  it("renders properly", () => {
    const component = shallow(<PermitResultsTable {...reducerProps} />);
    expect(component).toMatchSnapshot();
  });
});
