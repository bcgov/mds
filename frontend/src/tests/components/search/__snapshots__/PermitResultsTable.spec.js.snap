// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermitResultsTable renders properly 1`] = `
<div>
  <h2>
    Permits
  </h2>
  <Divider />
  <Table
    align="left"
    bordered={false}
    childrenColumnName="children"
    className="nested-table padding-large--bottom"
    columns={
      Array [
        Object {
          "dataIndex": "permit_guid",
          "key": "permit_guid",
          "render": [Function],
          "title": "Permit Guid",
        },
      ]
    }
    dataSource={
      Array [
        Object {
          "mine_guid": "5bf528d0-9e5a-421e-92c5-7e088374dad5",
          "mine_name": "Wheeler Inc",
          "permit_guid": "9df4593a-4b58-4cee-8c3a-1cad5df7d3f6",
          "permit_no": "CX-803",
          "permitee": "Carl Taylor, Goodwin and Wade",
        },
        Object {
          "mine_guid": "d5aeba94-6d58-4617-b71e-b16a145de86e",
          "mine_name": "Reese, Brown and Diaz",
          "permit_guid": "b066a984-55fc-47aa-8ecf-fb3bf33ec2bb",
          "permit_no": "CX-7230",
          "permitee": "Traci Franklin-Baker",
        },
        Object {
          "mine_guid": "af9b3e68-981e-42cb-9931-34f37d006378",
          "mine_name": "Mcgrath and Sons",
          "permit_guid": "88280bbd-18ab-483d-94d4-94e3063622f1",
          "permit_no": "CX-4926",
          "permitee": "Paul Parker, Mckee and Wilson",
        },
        Object {
          "mine_guid": "5392aef5-9a7b-4a94-98cd-66582fe423cb",
          "mine_name": "Brown-Wheeler",
          "permit_guid": "38e01cc1-bc49-430b-9453-1df508b98e64",
          "permit_no": "CX-9917",
          "permitee": "Nathaniel Harris-Randolph",
        },
        Object {
          "mine_guid": "74b668b3-0bce-4886-a93f-25fb59783ed3",
          "mine_name": "Williams-Baird",
          "permit_guid": "db1cbe69-1f13-4be3-a874-0093af1e6a49",
          "permit_no": "CX-7566",
          "permitee": "Jeffery Jones, Benson and Frazier",
        },
        Object {
          "mine_guid": "25003e54-314e-492e-b858-5af18381c6d3",
          "mine_name": "Harmon and Sons",
          "permit_guid": "ab74c658-891f-4a56-9604-fa4ba694ae73",
          "permit_no": "CX-63820",
          "permitee": "Denise Arroyo-Moran",
        },
      ]
    }
    indentSize={20}
    loading={false}
    locale={Object {}}
    pagination={false}
    rowKey="key"
    showHeader={false}
    size="default"
    sortDirections={
      Array [
        "ascend",
        "descend",
      ]
    }
    useFixedHeader={false}
  />
</div>
`;
